kernel/sched/rt.c

```diff
@@ -142,10 +142,12 @@ void free_rt_sched_group(struct task_group *tg)
 		destroy_rt_bandwidth(&tg->rt_bandwidth);
 
 	for_each_possible_cpu(i) {
-		if (tg->rt_rq)
-			kfree(tg->rt_rq[i]);
-		if (tg->rt_se)
-			kfree(tg->rt_se[i]);
+		/* Don't need to check if tg->rt_rq[i]
```

Comment:
It was actually checking if tg->rt_rq and tg->rt_se are NULL or not, not the elements in the array.

Reply:
I don't know how I managed to overlook that. I've received feedback from Peter Zijlstra describing this patch as "horrible" and I agree with that assessment. I'm waiting for a response from Peter regarding whether he thinks any part of the patch is salvageable or if the whole thing is better off being thrown out.